test/serialisation.jl

```diff
@@ -1,44 +1,111 @@
-using DFTK
-using JLD2
-using JSON3
-using WriteVTK
-using MPI
-import DFTK: ScfDefaultCallback, ScfSaveCheckpoints
-using Test
-include("testcases.jl")
-
-function test_scfres_agreement(tested, ref)
-    @test tested.basis.model.lattice           == ref.basis.model.lattice
-    @test tested.basis.model.temperature       == ref.basis.model.temperature
-    @test tested.basis.model.smearing          == ref.basis.model.smearing
-    @test tested.basis.model.εF                == ref.basis.model.εF
-    @test tested.basis.model.symmetries        == ref.basis.model.symmetries
-    @test tested.basis.model.spin_polarization == ref.basis.model.spin_polarization
-
-    @test tested.basis.model.positions == ref.basis.model.positions
-    @test atomic_symbol.(tested.basis.model.atoms) == atomic_symbol.(ref.basis.model.atoms)
-
-    @test tested.basis.Ecut      == ref.basis.Ecut
-    @test tested.basis.kweights  == ref.basis.kweights
-    @test tested.basis.fft_size  == ref.basis.fft_size
-
-    kcoords_test = getproperty.(tested.basis.kpoints, :coordinate)
-    kcoords_ref  = getproperty.(ref.basis.kpoints, :coordinate)
-    @test kcoords_test == kcoords_ref
-
-    # Note: Apart from the energy (which is recomputed on loading) the other quantities
-    #       should be exactly as stored. For MPI runs it is possible that the density differs
-    #       slightly for non-master processes as the version from the master process is stored.
-    @test tested.n_iter         == ref.n_iter
-    @test tested.energies.total ≈  ref.energies.total atol=1e-13
-    @test tested.eigenvalues    == ref.eigenvalues
-    @test tested.occupation     == ref.occupation
-    @test tested.ψ              == ref.ψ
-    @test tested.ρ              ≈  ref.ρ rtol=1e-14
+@testsetup module Serialisation
```

Comment:
Can we do two Testsetup modules here ?

Reply:
Well, if it leads to code duplication, than better have one module.

Reply:
Yeah that's really not pretty either.

Reply:
Fair enough. Maybe we'll have to bite some bullets in the end.

Reply:
Could you make a PR at TestItemRunner (e.g. in their docs) discussing this feature. Because it might be that this just "works" but is not intended behaviour. In that case I'm afraid we might have our tests randomly break in the future.

If they sort of give it a green light by admitting it to their public documentation, then I'm ok with using this here.